pkg/agent/status_manager.go

```diff
@@ -149,3 +167,15 @@ func (mgr *Manager) updateClusterStatus(ctx context.Context, namespace, clusterI
 		klog.WarningDepth(2, "failed to update status of ManagedCluster: %v", lastError)
 	}
 }
+
+func (mgr *Manager) getNewManagedClusterLabels() map[string]string {
+	originLabels := mgr.managedCluster.GetLabels()
+	modifiedLabels := map[string]string{}
```

Comment:
No need to introduce `modifiedLabels`.
We can make changes on `originLabels` directly.